splink/comparison_level_composition.py

```diff
@@ -0,0 +1,262 @@
+        m_probability (float, optional): Starting value for m probability.
+            Defaults to None.
+
+    Examples:
+        >>> import splink.duckdb.duckdb_comparison_level_library as cll
+        >>> # *Not* a null on first name `first_name`
+        >>> cll.cl_not(cll.exact_match("first_name"))
+
+        >>> import splink.duckdb.duckdb_comparison_level_library as cll
+        >>> # Find all exact matches *not* on the first of January
+        >>> dob_first_jan =  {
+        >>>    "sql_condition": "SUBSTR(dob_std_l, -5) = '01-01'",
+        >>>    "label_for_charts": "Date is 1st Jan",
+        >>> }
+        >>> exact_match_not_first_jan = cll.cl_and(
+        >>>     cll.exact_match_level("dob"),
+        >>>     cll.cl_not(dob_first_jan),
+        >>>     label_for_charts = "Exact match and not the 1st Jan"
+        >>> )
+
+
+    Returns:
+        ComparisonLevel
+            A new ComparisonLevel with the negated SQL condition and label_for_charts
+    """
+    dicts, sql_dialect = _parse_comparison_levels(cll)
+    result = {}
+    cld = dicts[0]
+    result["sql_condition"] = f"NOT ({cld['sql_condition']})"
+
+    result["label_for_charts"] = (
+        label_for_charts if label_for_charts else f"NOT ({_label_for_charts(cld)})"
+    )
+
+    if m_probability:
+        result["m_probability"] = m_probability
+
+    return ComparisonLevel(result, sql_dialect=sql_dialect)
+
+
+def _cl_merge(
+    *clls: ComparisonLevel | dict,
+    clause: str,
+    label_for_charts=None,
+    m_probability=None,
+    is_null_level=None,
+) -> ComparisonLevel:
+    if len(clls) == 0:
+        raise ValueError("Must provide at least one ComparisonLevel")
+
+    dicts, sql_dialect = _parse_comparison_levels(*clls)
+    result = {}
+    conditions = ("(" + d["sql_condition"] + ")" for d in dicts)
+    result["sql_condition"] = f" {clause} ".join(conditions)
+
+    # Set to null level if all supplied levels are "null levels"
+    if is_null_level is None:
+        if all([d.setdefault("is_null_level", False) for d in dicts]):
+            result["is_null_level"] = True
+
+    if label_for_charts:
+        result["label_for_charts"] = label_for_charts
+    else:
+        labels = ("(" + _label_for_charts(d) + ")" for d in dicts)
+        result["label_for_charts"] = f" {clause} ".join(labels)
+
+    if m_probability:
+        result["m_probability"] = m_probability
+
+    return ComparisonLevel(result, sql_dialect=sql_dialect)
+
+
+def _label_for_charts(comparison_dict: dict) -> str:
+    backup = comparison_dict["sql_condition"]
+    label = comparison_dict.get("label_for_charts", backup)
+
+    colname = comparison_dict.get("column_name")
+    if colname is None:
+        return label
+    # if null level, prefix with the column name
+    if comparison_dict.get("is_null_level", False):
+        label = f"{colname} is {label.upper()}"
+    # if exact match, suffix w/ colname
```

Comment:
So, this was meant to be a one-off here to aid with clarity.

In our basic comparison levels, null and exact match don't really need the column name appended as they're relatively clear as is.

I'm happy to move this across to the main CL class if you think it would aid readability for the charts.